scripts/build-testing-sdk.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * This script bundles some scripts used for testing:
+ * - amd loader
+ * - polyfills
+ *   + fetch
+ *   + promises
+ *   + bace64 encoding
+ * - then deps for the testing framework
+ */
+const fs = require('fs');
+
+const sourceModules = getSourceFromModules([
+  // umd module loader
+  'loader.js',
+  // original testing sdk deps
+  'route-recognizer',
+  'fake-xml-http-request',
+  'pretender'
+]);
+
+const outSource = `
+${sourceModules[0]}
+(function (self) {
+;self.fetch = null;
+${sourceModules.slice(1).join('\n')}
+}(window));
+`;
+
+// output to stdout so that we can
+// easily modify the output/output location via cli
+console.log(outSource);
+
+function getSourceFromModules(modules) {
```

Comment:
I find it really hard to read when you're using a function before its declaration. It makes a lot more sense to me to read a procedural module as a procedure, where the actual program executes sequentially, and the end of the script is at the end.